identity/aziot-identityd/src/lib.rs

```diff
@@ -379,14 +379,24 @@ impl auth::authorization::Authorizer for SettingsAuthorizer
 			crate::auth::OperationType::GetModule(m) => {
 				if let crate::auth::AuthId::LocalPrincipal(creds) = o.auth_id {
 					if let Some(p) = self.pmap.get(&crate::auth::Uid(creds.0)) {
-						return Ok(p.name.0 == m && p.id_type == Some(aziot_identity_common::IdType::Module))
+						let allow_id_type = match &p.id_type {
+							Some(i) => i.contains(&aziot_identity_common::IdType::Module),
+							None => false,
+						};
```

Comment:
More idiomatically, this is better (for this instance and other instances below) - 
```
let allow_id_type = p.id_type.clone()
    .map_or(false, |i| i.contains(&aziot_identity_common::IdType::Module));
```

Reply:
Changed